rotkehlchen/db/accounting_rules.py

```diff
@@ -117,30 +131,85 @@ def update_accounting_rule(
                     ' but it was not found',
                 )
 
+    def add_overwritten_setting(
+            self,
+            rule_identifier: int,
+            rule_property: Literal['count_entire_amount_spend', 'count_cost_basis_pnl'],
+            setting_name: Literal['include_gas_costs', 'include_crypto2crypto'],
+            is_active: bool = True,
+    ) -> None:
+        """
+        Store in the database a rule overwritten property that links it to a setting in the
+        settings table.
+        May raise: InputError
+        """
+        with self.db.conn.write_ctx() as write_cursor:
+            try:
+                write_cursor.execute(
+                    'INSERT INTO linked_rules_properties(accounting_rule, property_name, '
+                    'setting_name, active) VALUES (?, ?, ?, ?)',
+                    (rule_identifier, rule_property, setting_name, is_active),
+                )
+            except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
+                raise InputError('Overwritting rule already exists') from e
+
     def query_rules(
             self,
             filter_query: AccountingRulesFilterQuery,
-    ) -> tuple[
-        list[tuple[tuple[int, HistoryEventType, HistoryEventSubType, Optional[str]], BaseEventSettings]],  # noqa: E501
-        int,
-    ]:
+    ) -> tuple[list[RuleInformation], int]:
```

Comment:
adjust the docstring here

Reply:
I believe it can stay since esentially is the same

Reply:
you are right. I saw it all get back to one line and thought the second part of the tuple, the int was removed.